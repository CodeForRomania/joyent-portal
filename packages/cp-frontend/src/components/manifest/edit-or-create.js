import React, { Component } from 'react';
import { Field } from 'redux-form';
import styled from 'styled-components';
import SimpleTable from 'react-simple-table';
import { Row, Col } from 'react-styled-flexboxgrid';
import { Dots2 } from 'styled-text-spinners';
import Bundle from 'react-bundle';
import remcalc from 'remcalc';
import forceArray from 'force-array';

import {
  FormGroup,
  FormMeta,
  Input,
  Button,
  Card,
  Progressbar,
  ProgressbarItem,
  ProgressbarButton,
  H3,
  typography
} from 'joyent-ui-toolkit';

const Dl = styled.dl`
  margin: ${remcalc(13)} ${remcalc(19)};
`;

const ServiceName = H3.extend`
  margin-top: 0;
  margin-bottom: 0;
  line-height: 1.6;
  font-weight: 600;
`;

const ServiceCard = Card.extend`
  min-height: ${remcalc(72)};
`;

const ImageTitle = ServiceName.extend`
  display: inline-block;
`;

const Image = styled.span`
  ${typography.fontFamily};
`;

const ButtonsRow = Row.extend`
  margin-top: ${remcalc(29)};
  margin-bottom: ${remcalc(60)};
`;

const FilenameContainer = styled.span`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-content: stretch;
  align-items: stretch;
`;

const FilenameInput = styled(Input)`
  order: 0;
  flex: 1 1 auto;
  align-self: stretch;
`;

const FilenameRemove = Button.extend`
  order: 0;
  flex: 0 1 auto;
  align-self: auto;
  margin: ${remcalc(8)};
  margin-right: 0;
  height: ${remcalc(48)};
`;

class ManifestEditorBundle extends Component {
  constructor() {
    super();

    this.state = {};

    this.handleRender = this.handleRender.bind(this);
  }
  handleRender(ManifestEditor) {
    if (ManifestEditor) {
      setTimeout(() => {
        this.setState({ ManifestEditor });
      }, 80);
    }

    return <Dots2 />;
  }
  render() {
    if (!this.state.ManifestEditor) {
      return (
        <Bundle load={() => import('joyent-manifest-editor')}>
          {this.handleRender}
        </Bundle>
      );
    }

    const { ManifestEditor } = this.state;
    const { children, ...rest } = this.props;

    return (
      <ManifestEditor {...rest}>
        {children}
      </ManifestEditor>
    );
  }
}

const MEditor = ({ input, defaultValue }) =>
  <ManifestEditorBundle
    mode="yaml"
    {...input}
    value={input.value || defaultValue}
  />;

const EEditor = ({ input, defaultValue }) =>
  <ManifestEditorBundle
    mode="ini"
    {...input}
    value={input.value || defaultValue}
  />;

export const Name = ({ handleSubmit, onCancel, dirty }) =>
  <form onSubmit={handleSubmit}>
    <Row>
      <Col xs={12} md={3} lg={3}>
        <FormGroup name="name" reduxForm>
          <FormMeta left />
          <Input type="text" />
        </FormGroup>
      </Col>
    </Row>
    <ButtonsRow>
      <Button onClick={onCancel} secondary>
        Cancel
      </Button>
      <Button type="submit" disabled={!dirty}>
        Next
      </Button>
    </ButtonsRow>
  </form>;

export const Manifest = ({
  handleSubmit,
  onCancel,
  dirty,
  defaultValue = '',
  loading
}) =>
  <form onSubmit={handleSubmit}>
    <Field name="manifest" defaultValue={defaultValue} component={MEditor} />
    <ButtonsRow>
      <Button onClick={onCancel} secondary>
        Cancel
      </Button>
      <Button
        disabled={!(dirty || !loading || defaultValue.length)}
        type="submit"
      >
        Environment
      </Button>
    </ButtonsRow>
  </form>;

const Filename = ({ name, onRemoveFile }) =>
  <FilenameContainer>
    <FilenameInput
      type="text"
      placeholder="Filename including extensionâ€¦"
      defaultValue={name}
    />
    <FilenameRemove type="button" onClick={onRemoveFile} secondary>
      Remove
    </FilenameRemove>
  </FilenameContainer>;

export const Files = ({ loading, files, onRemoveFile }) => {
  if (loading) {
    return null;
  }

  const _files = files.map(({ id, name, value }) =>
    <div key={id}>
      <FormGroup name={`file-name-${id}`} reduxForm>
        <FormMeta left />
        <Filename name={name} onRemoveFile={() => onRemoveFile(id)} />
      </FormGroup>
      <Field
        name={`file-value-${id}`}
        defaultValue={value}
        component={EEditor}
      />
    </div>
  );

  return (
    <div>
      <H3>Files:</H3>
      {_files}
    </div>
  );
};

export const Environment = ({
  handleSubmit,
  onCancel,
  onAddFile,
  onRemoveFile,
  dirty,
  defaultValue = '',
  files = [],
  loading
}) =>
  <form onSubmit={handleSubmit}>
    <Field name="environment" defaultValue={defaultValue} component={EEditor} />
    <Files files={files} onRemoveFile={onRemoveFile} loading={loading} />
    <ButtonsRow>
      <Button onClick={onCancel} secondary>
        Cancel
      </Button>
      <Button type="button" onClick={onAddFile} secondary>
        Add File
      </Button>
      <Button
        disabled={!(dirty || !loading || defaultValue.length)}
        type="submit"
      >
        {loading ? <Dots2 /> : 'Review'}
      </Button>
    </ButtonsRow>
  </form>;

export const Review = ({ handleSubmit, onCancel, dirty, ...state }) => {
  const serviceList = forceArray(state.services).map(({ name, config }) =>
    <ServiceCard key={name}>
      <Dl>
        <dt>
          <ServiceName>
            {name}
          </ServiceName>
        </dt>
        <dt>
          <ImageTitle>Image:</ImageTitle> <Image>{config.image}</Image>
        </dt>
        {config.environment.length
          ? <dt>
              <ImageTitle>Environment:</ImageTitle>
            </dt>
          : undefined}
        {config.environment.length
          ? <SimpleTable
              columns={[
                {
                  columnHeader: 'Name',
                  path: 'name'
                },
                {
                  columnHeader: 'Value',
                  path: 'value'
                }
              ]}
              data={config.environment}
            />
          : undefined}
      </Dl>
    </ServiceCard>
  );

  return (
    <form onSubmit={handleSubmit}>
      {serviceList}
      <ButtonsRow>
        <Button onClick={onCancel} disabled={state.loading} secondary>
          Cancel
        </Button>
        <Button disabled={state.loading} type="submit">
          {state.loading ? <Dots2 /> : 'Confirm and Deploy'}
        </Button>
      </ButtonsRow>
    </form>
  );
};

export const Progress = ({ stage, create, edit }) => {
  const _nameCompleted = stage !== 'name';
  const _nameActive = stage === 'name';

  const _name = !create
    ? null
    : <ProgressbarItem>
        <ProgressbarButton
          zIndex="10"
          completed={_nameCompleted}
          active={_nameActive}
          first
        >
          Name the group
        </ProgressbarButton>
      </ProgressbarItem>;

  const _manifestCompleted = ['environment', 'review'].indexOf(stage) >= 0;
  const _manifestActive = create ? stage === 'manifest' : stage === 'edit';

  const _manifest = (
    <ProgressbarItem>
      <ProgressbarButton
        zIndex="9"
        completed={_manifestCompleted}
        active={_manifestActive}
        first={edit}
      >
        Define Services
      </ProgressbarButton>
    </ProgressbarItem>
  );

  const _environmentCompleted = stage === 'review';
  const _environmentActive = stage === 'environment';

  const _environment = (
    <ProgressbarItem>
      <ProgressbarButton
        zIndex="8"
        completed={_environmentCompleted}
        active={_environmentActive}
      >
        Define Environment
      </ProgressbarButton>
    </ProgressbarItem>
  );

  const _reviewActive = stage === 'review';

  const _review = (
    <ProgressbarItem>
      <ProgressbarButton zIndex="7" active={_reviewActive} last>
        Review and deploy
      </ProgressbarButton>
    </ProgressbarItem>
  );

  return (
    <Progressbar>
      {_name}
      {_manifest}
      {_environment}
      {_review}
    </Progressbar>
  );
};
