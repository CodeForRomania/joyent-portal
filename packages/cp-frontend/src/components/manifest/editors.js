import React from 'react';
import ManifestEditorBundle from './manifest-editor';

export const MEditor = ({ input, defaultValue, readOnly }) => (
  <ManifestEditorBundle
    mode="yaml"
    {...input}
    value={input.value || defaultValue}
    readOnly={readOnly}
  />
);

export const EEditor = ({ input, defaultValue, readOnly }) => (
  <ManifestEditorBundle
    mode="ini"
    {...input}
    value={input.value || defaultValue}
    readOnly={readOnly}
  />
);
