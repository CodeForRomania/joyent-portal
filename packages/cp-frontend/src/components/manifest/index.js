export { default as Files } from './files';
export { default as Editors } from './editors';
export { default as ManifestEditorBundle } from './manifest-editor';
export { default as Manifest } from './manifest';
export { default as Name } from './name';
export { default as Review } from './review';
export { default as Progress } from './progress';
export { default as Environment } from './environment';