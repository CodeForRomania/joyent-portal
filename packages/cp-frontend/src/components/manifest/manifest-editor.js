import React, { Component } from 'react';
import Bundle from 'react-bundle';

import { Loader } from '@components/messaging';

class ManifestEditorBundle extends Component {
  constructor() {
    super();

    this.state = {};

    this.handleRender = this.handleRender.bind(this);
  }
  handleRender(ManifestEditor) {
    if (ManifestEditor) {
      setTimeout(() => {
        this.setState({ ManifestEditor });
      }, 80);
    }

    return <Loader />;
  }
  render() {
    if (!this.state.ManifestEditor) {
      return (
        <Bundle load={() => import('joyent-manifest-editor')}>
          {this.handleRender}
        </Bundle>
      );
    }

    const { ManifestEditor } = this.state;
    const { children, ...rest } = this.props;

    return <ManifestEditor {...rest}>{children}</ManifestEditor>;
  }
}

export default ManifestEditorBundle;
