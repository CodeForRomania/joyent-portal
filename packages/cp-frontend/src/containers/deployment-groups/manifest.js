import React from 'react';
import { compose, graphql } from 'react-apollo';
import get from 'lodash.get';

import ManifestQuery from '@graphql/Manifest.gql';
import DeploymentGroupBySlugQuery from '@graphql/DeploymentGroupBySlug.gql';

import DeploymentGroupEditOrCreate from './edit-or-create';
import { Progress } from '@components/deployment-groups/create';
import { LayoutContainer } from '@components/layout';
import { DeploymentGroupsLoading } from '@components/deployment-groups';
import { H2 } from 'joyent-ui-toolkit';

const Manifest = ({
  loading,
  error,
  manifest = '',
  deploymentGroup = null,
  match
}) => {
  const stage = match.params.stage;
  const _loading = !loading ? null : <DeploymentGroupsLoading />;
  const _error = !error ? null : <span>{error.toString()}</span>;

  const _view = loading || !deploymentGroup
    ? null
    : <DeploymentGroupEditOrCreate
        edit
        manifest={manifest}
        deploymentGroup={deploymentGroup}
      />;

  const _notice = !error &&
    !loading &&
    deploymentGroup &&
    deploymentGroup.imported &&
    !manifest
    ? <span>
        Since this DeploymentGroup was imported, it doesn't have the initial
        manifest
      </span>
    : null;

  return (
    <LayoutContainer>
      <H2>Edit Manifest</H2>
      <Progress stage={stage} edit />
      {_error}
      {_loading}
      {_notice}
      {_view}
    </LayoutContainer>
  );
};

export default compose(
  graphql(ManifestQuery, {
    options: props => ({
      variables: {
        deploymentGroupSlug: props.match.params.deploymentGroup
      }
    }),
    props: ({ data: { deploymentGroup, loading, error } }) => ({
      manifest: get(deploymentGroup, 'version.manifest.raw', ''),
      loading,
      error
    })
  }),
  graphql(DeploymentGroupBySlugQuery, {
    options: props => ({
      variables: {
        slug: props.match.params.deploymentGroup
      }
    }),
    props: ({ data: { deploymentGroups, loading, error } }) => ({
      deploymentGroup: deploymentGroups && deploymentGroups.length
        ? deploymentGroups[0]
        : null,
      loading,
      error
    })
  })
)(Manifest);
