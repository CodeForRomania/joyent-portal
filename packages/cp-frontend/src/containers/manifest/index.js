import React from 'react';
import { compose, graphql } from 'react-apollo';
import get from 'lodash.get';
import forceArray from 'force-array';

import ManifestQuery from '@graphql/Manifest.gql';
import DeploymentGroupBySlugQuery from '@graphql/DeploymentGroupBySlug.gql';

import ManifestEditOrCreate from '@containers/manifest/edit-or-create';
import { Progress } from '@components/manifest/edit-or-create';
import { LayoutContainer } from '@components/layout';
import { Title } from '@components/navigation';
import { Loader, ErrorMessage, WarningMessage } from '@components/messaging';

const Manifest = ({
  loading,
  error,
  manifest = '',
  environment = '',
  files = [],
  deploymentGroup = null,
  hasManifest = false,
  match
}) => {
  const stage = match.params.stage;
  const _title = <Title>Edit Manifest</Title>;

  if (loading || !deploymentGroup || !hasManifest) {
    return (
      <LayoutContainer center>
        {_title}
        <Loader />
      </LayoutContainer>
    );
  }

  if (error) {
    return (
      <LayoutContainer>
        {_title}
        <ErrorMessage
          title="Ooops!"
          message="An error occurred while loading your deployment group."
        />
      </LayoutContainer>
    );
  }

  const _notice =
    deploymentGroup && deploymentGroup.imported && !manifest ? (
      <WarningMessage
        title="Be aware"
        message="Since this Deployment Group was imported, it doesn&#x27;t have the initial manifest."
      />
    ) : null;

  return (
    <LayoutContainer>
      {_title}
      <Progress stage={stage} edit />
      {_notice}
      <ManifestEditOrCreate
        manifest={manifest}
        environment={environment}
        files={files}
        deploymentGroup={deploymentGroup}
        edit
      />
    </LayoutContainer>
  );
};

export default compose(
  graphql(ManifestQuery, {
    options: props => ({
      fetchPolicy: 'network-only',
      variables: {
        deploymentGroupSlug: props.match.params.deploymentGroup
      }
    }),
    props: ({ data: { deploymentGroup, loading, error } }) => ({
      files: get(deploymentGroup, 'version.manifest.files', []),
      manifest: get(deploymentGroup, 'version.manifest.raw', ''),
      environment: get(deploymentGroup, 'version.manifest.environment', ''),
      hasManifest: Boolean(get(deploymentGroup, 'version.manifest')),
      loading,
      error
    })
  }),
  graphql(DeploymentGroupBySlugQuery, {
    options: props => ({
      variables: {
        slug: props.match.params.deploymentGroup
      }
    }),
    props: ({
      data: { deploymentGroups, loading, error, startPolling, stopPolling }
    }) => {
      const dgs = forceArray(deploymentGroups);

      if (!dgs.length) {
        startPolling(1000);
      } else {
        stopPolling();
      }

      return {
        deploymentGroup: dgs[0],
        loading,
        error
      };
    }
  })
)(Manifest);
