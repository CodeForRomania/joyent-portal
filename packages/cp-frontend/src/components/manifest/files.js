import React from 'react';
import { Field } from 'redux-form';
import styled from 'styled-components';
import remcalc from 'remcalc';
import { EEditor } from './editors';

import {
  FormGroup,
  Input,
  Button,
  Card
} from 'joyent-ui-toolkit';

const FilenameContainer = styled.span`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-content: stretch;
  align-items: stretch;
`;

const FilenameInput = styled(Input)`
  order: 0;
  flex: 1 1 auto;
  align-self: stretch;
  margin: 0 0 ${remcalc(13)} 0;
`;

const FilenameRemove = Button.extend`
  order: 0;
  flex: 0 1 auto;
  align-self: auto;
  margin: 0 0 0 ${remcalc(8)};
  height: ${remcalc(48)};
`;

const FileCard = Card.extend`padding: ${remcalc(24)} ${remcalc(19)};`;

const File = ({ id, name, value, onRemoveFile, readOnly }) => {
  const removeButton = !readOnly ? (
    <FilenameRemove type="button" onClick={onRemoveFile} secondary>
      Remove
    </FilenameRemove>
  ) : null;

  const fileEditor = !readOnly ? (
    <Field name={`file-value-${id}`} defaultValue={value} component={EEditor} />
  ) : (
    <EEditor input={{ value }} readOnly />
  );

  const input = !readOnly ? (
    <FilenameInput type="text" placeholder="Filename including extension…" />
  ) : (
    <FilenameInput
      type="text"
      placeholder="Filename including extension…"
      value={name}
    />
  );

  return (
    <FileCard>
      <FormGroup name={`file-name-${id}`} reduxForm={!readOnly}>
        <FilenameContainer>
          {input}
          {removeButton}
        </FilenameContainer>
      </FormGroup>
      {fileEditor}
    </FileCard>
  );
};

const Files = ({ files, onAddFile, onRemoveFile, readOnly }) => {
  const footer = !readOnly ? (
    <Button type="button" onClick={onAddFile} secondary>
      Create new .env file
    </Button>
  ) : null;

  return (
    <div>
      {files.map(({ id, ...rest }) => (
        <File
          key={id}
          id={id}
          onRemoveFile={() => onRemoveFile(id)}
          readOnly={readOnly}
          {...rest}
        />
      ))}
      {footer}
    </div>
  );
};

export default Files;