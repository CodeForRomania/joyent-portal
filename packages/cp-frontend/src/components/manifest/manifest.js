import React from 'react';
import { FormMeta, Button, Label } from 'joyent-ui-toolkit';
import { Row } from 'react-styled-flexboxgrid';
import remcalc from 'remcalc';
import { Field } from 'redux-form';
import { MEditor } from './editors';

const ButtonsRow = Row.extend`margin: ${remcalc(29)} 0 ${remcalc(60)} 0;`;

export const Manifest = ({
  handleSubmit,
  onCancel,
  dirty,
  defaultValue = '',
  loading
}) => (
  <form onSubmit={handleSubmit}>
    <FormMeta left>
      <Label>Project manifest</Label>
    </FormMeta>
    <Field name="manifest" defaultValue={defaultValue} component={MEditor} />
    <ButtonsRow>
      <Button type="button" onClick={onCancel} secondary>
        Cancel
      </Button>
      <Button
        disabled={!(dirty || !loading || defaultValue.length)}
        loading={loading}
        type="submit"
      >
        Environment
      </Button>
    </ButtonsRow>
  </form>
);

export default Manifest;