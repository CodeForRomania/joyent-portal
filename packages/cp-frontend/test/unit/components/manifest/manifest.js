/**
 * @jest-environment jsdom
 */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import 'jest-styled-components';
import { Store } from '../../mocks';
import { reduxForm } from 'redux-form';

import { Manifest } from '@components/manifest';

const ManifestTestForm = reduxForm({ form: 'ManifestTestForm' })(Manifest);

it('renders <Manifest /> without throwing', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <Store>
      <ManifestTestForm
        defaultValue="test"
        handleSubmit={() => {}}
        onCancel={() => {}}
        dirty
        loading
      />
    </Store>
  );
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it('renders <Manifest /> without throwing', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <Store>
        <ManifestTestForm
          defaultValue="test"
          handleSubmit={() => {}}
          onCancel={() => {}}
          loading
        />
      </Store>
    )
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

it('renders <Manifest /> without throwing', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <Store>
        <ManifestTestForm
          defaultValue="test"
          handleSubmit={() => {}}
          onCancel={() => {}}
        />
      </Store>
    )
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});