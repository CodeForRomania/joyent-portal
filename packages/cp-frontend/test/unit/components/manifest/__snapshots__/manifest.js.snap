// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Manifest /> without throwing 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "addTypename": true,
      "dataId": [Function],
      "dataIdFromObject": [Function],
      "disableNetworkFetches": false,
      "fieldWithArgs": [Function],
      "fragmentMatcher": HeuristicFragmentMatcher {},
      "initialState": Object {},
      "middleware": [Function],
      "mutate": [Function],
      "networkInterface": HTTPFetchNetworkInterface {
        "_afterwares": Array [],
        "_middlewares": Array [],
        "_opts": Object {},
        "_uri": "https://localhost:443/api/graphql",
      },
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "addTypename": true,
        "deduplicator": Deduplicator {
          "inFlightRequestPromises": Object {},
          "networkInterface": HTTPFetchNetworkInterface {
            "_afterwares": Array [],
            "_middlewares": Array [],
            "_opts": Object {},
            "_uri": "https://localhost:443/api/graphql",
          },
        },
        "disableBroadcasting": false,
        "fetchQueryPromises": Object {},
        "fragmentMatcher": HeuristicFragmentMatcher {},
        "idCounter": 1,
        "lastRequestId": Object {},
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "networkInterface": HTTPFetchNetworkInterface {
          "_afterwares": Array [],
          "_middlewares": Array [],
          "_opts": Object {},
          "_uri": "https://localhost:443/api/graphql",
        },
        "observableQueries": Object {},
        "pollingTimers": Object {},
        "queryDeduplication": true,
        "queryDocuments": Object {},
        "queryIdsByName": Object {},
        "queryListeners": Object {},
        "queryStore": QueryStore {
          "store": Object {},
        },
        "reducerConfig": Object {
          "addTypename": true,
          "customResolvers": undefined,
          "dataIdFromObject": [Function],
          "fragmentMatcher": [Function],
        },
        "reduxRootSelector": [Function],
        "scheduler": QueryScheduler {
          "inFlightQueries": Object {},
          "intervalQueries": Object {},
          "pollingTimers": Object {},
          "queryManager": [Circular],
          "registeredQueries": Object {},
        },
        "ssrMode": false,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
        },
      },
      "reducerConfig": Object {
        "addTypename": true,
        "customResolvers": undefined,
        "dataIdFromObject": [Function],
        "fragmentMatcher": [Function],
      },
      "resetStore": [Function],
      "setStore": [Function],
      "ssrMode": false,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
      },
      "version": "1.9.2",
      "watchQuery": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ReduxForm
    defaultValue="test"
    dirty={true}
    handleSubmit={[Function]}
    loading={true}
    onCancel={[Function]}
  />
</ApolloProvider>
`;

exports[`renders <Manifest /> without throwing 2`] = `
<ApolloProvider
  client={
    ApolloClient {
      "addTypename": true,
      "dataId": [Function],
      "dataIdFromObject": [Function],
      "disableNetworkFetches": false,
      "fieldWithArgs": [Function],
      "fragmentMatcher": HeuristicFragmentMatcher {},
      "initialState": Object {},
      "middleware": [Function],
      "mutate": [Function],
      "networkInterface": HTTPFetchNetworkInterface {
        "_afterwares": Array [],
        "_middlewares": Array [],
        "_opts": Object {},
        "_uri": "https://localhost:443/api/graphql",
      },
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "addTypename": true,
        "deduplicator": Deduplicator {
          "inFlightRequestPromises": Object {},
          "networkInterface": HTTPFetchNetworkInterface {
            "_afterwares": Array [],
            "_middlewares": Array [],
            "_opts": Object {},
            "_uri": "https://localhost:443/api/graphql",
          },
        },
        "disableBroadcasting": false,
        "fetchQueryPromises": Object {},
        "fragmentMatcher": HeuristicFragmentMatcher {},
        "idCounter": 1,
        "lastRequestId": Object {},
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "networkInterface": HTTPFetchNetworkInterface {
          "_afterwares": Array [],
          "_middlewares": Array [],
          "_opts": Object {},
          "_uri": "https://localhost:443/api/graphql",
        },
        "observableQueries": Object {},
        "pollingTimers": Object {},
        "queryDeduplication": true,
        "queryDocuments": Object {},
        "queryIdsByName": Object {},
        "queryListeners": Object {},
        "queryStore": QueryStore {
          "store": Object {},
        },
        "reducerConfig": Object {
          "addTypename": true,
          "customResolvers": undefined,
          "dataIdFromObject": [Function],
          "fragmentMatcher": [Function],
        },
        "reduxRootSelector": [Function],
        "scheduler": QueryScheduler {
          "inFlightQueries": Object {},
          "intervalQueries": Object {},
          "pollingTimers": Object {},
          "queryManager": [Circular],
          "registeredQueries": Object {},
        },
        "ssrMode": false,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
        },
      },
      "reducerConfig": Object {
        "addTypename": true,
        "customResolvers": undefined,
        "dataIdFromObject": [Function],
        "fragmentMatcher": [Function],
      },
      "resetStore": [Function],
      "setStore": [Function],
      "ssrMode": false,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
      },
      "version": "1.9.2",
      "watchQuery": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ReduxForm
    defaultValue="test"
    handleSubmit={[Function]}
    loading={true}
    onCancel={[Function]}
  />
</ApolloProvider>
`;

exports[`renders <Manifest /> without throwing 3`] = `
<ApolloProvider
  client={
    ApolloClient {
      "addTypename": true,
      "dataId": [Function],
      "dataIdFromObject": [Function],
      "disableNetworkFetches": false,
      "fieldWithArgs": [Function],
      "fragmentMatcher": HeuristicFragmentMatcher {},
      "initialState": Object {},
      "middleware": [Function],
      "mutate": [Function],
      "networkInterface": HTTPFetchNetworkInterface {
        "_afterwares": Array [],
        "_middlewares": Array [],
        "_opts": Object {},
        "_uri": "https://localhost:443/api/graphql",
      },
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "addTypename": true,
        "deduplicator": Deduplicator {
          "inFlightRequestPromises": Object {},
          "networkInterface": HTTPFetchNetworkInterface {
            "_afterwares": Array [],
            "_middlewares": Array [],
            "_opts": Object {},
            "_uri": "https://localhost:443/api/graphql",
          },
        },
        "disableBroadcasting": false,
        "fetchQueryPromises": Object {},
        "fragmentMatcher": HeuristicFragmentMatcher {},
        "idCounter": 1,
        "lastRequestId": Object {},
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "networkInterface": HTTPFetchNetworkInterface {
          "_afterwares": Array [],
          "_middlewares": Array [],
          "_opts": Object {},
          "_uri": "https://localhost:443/api/graphql",
        },
        "observableQueries": Object {},
        "pollingTimers": Object {},
        "queryDeduplication": true,
        "queryDocuments": Object {},
        "queryIdsByName": Object {},
        "queryListeners": Object {},
        "queryStore": QueryStore {
          "store": Object {},
        },
        "reducerConfig": Object {
          "addTypename": true,
          "customResolvers": undefined,
          "dataIdFromObject": [Function],
          "fragmentMatcher": [Function],
        },
        "reduxRootSelector": [Function],
        "scheduler": QueryScheduler {
          "inFlightQueries": Object {},
          "intervalQueries": Object {},
          "pollingTimers": Object {},
          "queryManager": [Circular],
          "registeredQueries": Object {},
        },
        "ssrMode": false,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
        },
      },
      "reducerConfig": Object {
        "addTypename": true,
        "customResolvers": undefined,
        "dataIdFromObject": [Function],
        "fragmentMatcher": [Function],
      },
      "resetStore": [Function],
      "setStore": [Function],
      "ssrMode": false,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
      },
      "version": "1.9.2",
      "watchQuery": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ReduxForm
    defaultValue="test"
    handleSubmit={[Function]}
    onCancel={[Function]}
  />
</ApolloProvider>
`;
